src/app/clusters/time-synchronization-server/time-synchronization-server.cpp

```diff
@@ -551,7 +551,7 @@ void TimeSynchronizationServer::InitTimeZone()
     for (auto & tzStore : mTimeZoneObj.timeZoneList)
     {
         memset(tzStore.name, 0, sizeof(tzStore.name));
-        tzStore.timeZone = { .offset = 0, .validAt = 0, .name = MakeOptional(CharSpan(tzStore.name, sizeof(tzStore.name))) };
+        tzStore.timeZone = { .offset = 0, .validAt = 0, .name = MakeOptional(CharSpan(tzStore.name, 0)) };
```

Comment:
Is this change from a different PR?

Reply:
Somewhat. Without this, the devices will try to send `{0x00, .... ` due to #30342, so I added this as a minimal change to not have our CI fail due to our now strict spec enforcing.

Reply:
The change is correct: if name is empty, set size to empty. But it is also incomplete and that will need to be fixed as a followup (#30343 patches to fix all sends, but still ugly. I belive @fessehaeve would follow up with a proper fix after the FC deadline in Nov 20)